### ISSUE TITLE
views on NamedArrays are slow

### ISSUE THREAD

**Issue by mkborregaard** [davidavdav/NamedArrays.jl#49]
Applying view on a NamedArray tends to be slow, probably because the dimnames Dicts are recalculated. I had a bit of trouble generating an MWE, but on a big (10.000 x 18.000) sparse Boolean NamedMatrix `a` I had `@benchmark view($a, :, :)` benchmark at 7 ms, while `@benchmark view($(a.array), :, :)` benchmarked at 20 ns, i.e. a speedup of a factor 300.000!

**Comment by davidavdav** [comment]
It looks like the code in `view(::NamedArray)` is a bit of a hack to get it going, not understanding much of the workings of `view()`.  
```julia
Base.view{T,N}(n::NamedArray{T,N}, I::Vararg{Any,N}) = namedgetindex(n, map((d,i)->indices(d, i), n.dicts, I)...; useview=true)
```
There is a `map()` going on, and further, `namedgetindex()` does quite a bit of logic to recompute the names of the sliced arrays. 

If view has to be efficient for NamedArrays, this would probably need a lot of rework.

**Comment by mkborregaard** [comment]
Thanks for the heads up! I think for my purposes I may refactor my package to use normal Arrays at the moment, as performance is important. I'm not even sure it can be made very efficient, as the key to the quick `namedgetindex` lies in the names Dicts being updated, right? Maybe it's simply a tradeoff of quick views vs quick named `getindex`.

**Comment by davidavdav** [comment]
I think the whole idea about a view is that only indices are recomputed in accessing individual elements in the view.  So `view(NamedArray(::Array))` should probably translate `NamedArray(view(::Array), view(dicts))` somehow.

**Comment by mkborregaard** [comment]
Yes absolutely, but the values in the dicts are indices that become invalidated by the view, right? Or am I misunderstanding the design?
